airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/dev_assets.py

```diff
@@ -164,3 +206,25 @@ def persist_metadata_definitions(context, overrode_metadata_definitions):
     file_paths_str = "\n".join(file_paths)
 
     return Output(files, metadata={"count": len(files), "file_paths": file_paths_str})
+
+
+@asset(group_name=GROUP_NAME)
+def cloud_catalog_diff(cloud_catalog_from_metadata: dict, latest_cloud_catalog_dict: dict):
+    """
+    Compares the cloud catalog from the metadata with the latest OSS catalog.
+    """
+    diff = diff_catalogs(latest_cloud_catalog_dict, cloud_catalog_from_metadata)
+    diff_df = pd.DataFrame.from_dict(diff.to_dict())
+
+    return OutputDataFrame(diff_df)
+
+
+@asset(group_name=GROUP_NAME)
+def oss_catalog_diff(oss_catalog_from_metadata: dict, latest_oss_catalog_dict: dict):
+    """
+    Compares the OSS catalog from the metadata with the latest OSS catalog.
+    """
+    diff = diff_catalogs(latest_oss_catalog_dict, oss_catalog_from_metadata)
+    diff_df = pd.DataFrame.from_dict(diff)
+
+    return OutputDataFrame(diff_df)
```

Comment:
Could you share why these assets return dataframe and not pure DeepDiff objects?

Reply:
Good question: I have it return a dataframe to take advantage of the `to_markdown` method for previews

Reply:
Would it make sense to provide a final asset only exposing the dataframe and depending on the oss_catalog_diff which returns the diff object? Might be overkill at the moment.

Reply:
It may slightly violate the "only write the code when you need to" rule 😛 

but its a good idea and what dagster is meant for so lets do it!